standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -1562,16 +1620,46 @@ public void markRefused(CompactionInfo info) throws MetaException {
 
   @Override
   @RetrySemantics.CannotRetry
-  public void setCleanerRetryRetentionTimeOnError(CompactionInfo info) throws MetaException {
+  public void setCleanerRetryRetentionTimeOnError(CompactionInfo info, boolean isAbort) throws MetaException {
```

Comment:
I had created a separate generic class from which CompactionInfo derives from in #4091 , however it was pulled out in review phase by @deniskuzZ . 
https://github.com/apache/hive/pull/4091#discussion_r1161666811

I will wait till @deniskuzZ advises on whether a separate class needs to be created or reuse CompactionInfo before fixing this.

Reply:
I am fine with AbortRequest, just try not to duplicate much

Reply:
Created AbortTxnRequestInfo which is extended from CompactionInfo. Created a separate API for setting retry retention for abort retry records called `setAbortCleanerRetryRetentionTimeOnError`.

Reply:
should we extend from CompactionInfo? I think, just e few fields are actually required for abort functionality

Reply:
I have removed AbortTxnRequestInfo since it is not required anymore due to the introduction of a separate type.
Having a separate type is very powerful since we can use existing APIs to handle all requests.